### ISSUE TITLE
Feature Request: Invisible lines

### ISSUE THREAD

**Issue by rogeralsing** [mermaid-js/mermaid#2805]
Controlling layouts in Mermaid is somewhat problematic.
Often we have to resort to strange hacks and CSS tricks to make the diagrams look as intended.

One way to achieve better control could be to introduce invisible lines.
just like normal lines, with the same layout rules, but just not visible.

e.g. in a real use-case for me here.
I would like to insert a shape where the red blob is:
<img width="685" alt="Skärmavbild 2022-03-08 kl  16 13 34" src="https://user-images.githubusercontent.com/647031/157266873-5039d75b-8262-4279-b488-d45eec7852c7.png">

But when adding new shapes they appear at the bottom, to the left.

If I could have an invisible line from "Exit" to the red blob, that would solve it in this case, as the layout engine would then place that block where I want it.

**Comment by rogeralsing** [comment]
Am I missing something?
I don't see such "missing" underscore lines at that line in the link

<img width="1013" alt="Skärmavbild 2022-06-07 kl  20 13 44" src="https://user-images.githubusercontent.com/647031/172453514-11d679df-34b0-4c77-bd39-bf848c1a6cc0.png">

**Comment by flywire** [comment]
> I don't see such "missing" underscore lines at that line in the link

No, it's a proposal for construction features used to calculate layout but not drawn. Could also work for nodes, groups etc.

**Comment by fkohrt** [comment]
Isn't this a duplicate of #3192?

**Comment by jsolbrig** [comment]
I like @flywire's suggestion of using underscores to denote an invisible or missing link. I prefer the term "invisible", though, because it describes the fact that there is a link there, it just isn't rendered.

I'm wondering, what is the difficult part of doing this? Is it simply that there is no way in Mermaid to create a relationship between nodes without actually drawing a line to connect them?

I would be happy to try implementing an update for this if I had an idea of where to start, but it would take me a while to even begin to wrap my head around the package to see where things would need to be changed... If someone could point me in the right direction on what might need to be changed or if this is even possible, I'd be happy to take a look.

**Comment by fkohrt** [comment]
@jsolbrig The issue #3192 that I previously mentioned refers to two commits that already implemented invisible links with v9.2.0, but the feature was removed again with v9.3.0. @knsv might know more about this. The commits in question are eec97d1 and 4d46ea9. But I guess the first step would be to find out why it got removed.

**Comment by jsolbrig** [comment]
@fkohrt Thank you for pointing that out! I hadn't noticed the two commits despite that being my own issue.

Those changes look simple enough. I wonder why they were not kept. Hopefully @knsv can fill us in.

**Comment by knsv** [comment]
Hi!

These are duplicates issues and the feature should never have been removed after 9.3.0. Sorry about that. 

This issue was created earlier then 3192 but I missed that, otherwise I would have committed the solution on this ticked instead of #3192.

In v10 invisisble lines will be re-introduced again. (Closing both the issues).

/Knut

**Comment by jsolbrig** [comment]
@knsv Thank you!

**Comment by fkohrt** [comment]
This issue has been resolved by #4113 and can be closed.

**Comment by Yokozuna59** [comment]
@fkohrt Thanks!
